File: easy/0258_add_digits.rb
class Solution

  def add_digits(num)
    n = num
    while n >= 10
      sum = 0
      while n > 0
        sum += n % 10
        n = n / 10
      end
      n = sum
    end

    return n
  end

end

p Solution.new().add_digits(38)

File: easy/0258_add_digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        n = num
        while n >= 10:
            summ = 0
            while n > 0:
                summ += n % 10
                n = int(n / 10)
            n = summ

        return n

print(Solution().addDigits(38)) # 3 + 8 = 11, 1 + 1 = 2, ans = 2